src/librustc_mir_build/hair/pattern/_match.rs

```diff
@@ -480,7 +480,9 @@ impl<'p, 'tcx> Matrix<'p, 'tcx> {
     /// Pushes a new row to the matrix. If the row starts with an or-pattern, this expands it.
     crate fn push(&mut self, row: PatStack<'p, 'tcx>) {
         if let Some(rows) = row.expand_or_pat() {
-            self.0.extend(rows);
+            for row in rows {
+                self.push(row)
```

Comment:
Maybe leave a comment to explain that we recursively expand the or-patterns of the new rows ? Otherwise it might not be clear that the distinction between `self.push` and `self.0.push` is on purpose

Reply:
Added a comment, how does that look?

Reply:
Perfect, thanks !